### DISCUSSION TITLE
Can't rename buttons in web UI

### DISCUSSION THREAD

**Discussion by justin-lake** [aqualinkd/AqualinkD#202] [General]
Just got Aqualink set up and most things are working as expected.  However, the button that reads "Spa Blower" in the UI controls my lights and I can't figure out which button actually controls the blower.  I've attempted to rename the buttons in the config file, but it's not showing a change in the UI.  I'm sure I'm missing something simple, but just can't figure this out.  Any suggestions?

![image](https://user-images.githubusercontent.com/35898351/229001788-bef0846a-9a46-4b88-96d8-f8ecac5e584c.png)

![image](https://user-images.githubusercontent.com/35898351/229001831-17c7c108-a42e-433b-ba93-276c8a8fc31e.png)

**Comment by tomtokic** [comment]
I have been using it for a while and it uses my PDA lables in the UI by having them set with the following tags:
button_01_PDA_label=FILTER PUMP

Try using that in your aqualinkd.conf file.

**Reply by justin-lake** [reply]
I don't have the PDA on my system, though.  Does that matter?

**Comment by sfeakes** [comment]
Aqualinkd uses heavy browser caching, clear you browsers cache or use different browser when you change anything to do with the web ui.

PDA labels will only make a difference if you have a PDA only control panel.

**Reply by justin-lake** [reply]
I'm definitely doing _something_ wrong.  No matter what I do with a fresh browser, clearing cache, etc. I still get the same button names.  In fact, just for grins, I commented out the button labels and everything is still staying the same, incorrectly.

I'm beginning to wonder if I'm editing the wrong .conf file, but I'm only finding a single instance of the file on a search.

**Comment by sfeakes** [comment]
File location will depend on where / how you installed aqualinkd.
/etc/aqualinkd.conf is default location.
You also need to restart Aqualinkd for the config to be read.

**Reply by justin-lake** [reply]
Thank you for weighing in @sfeakes.  I was somehow editing the wrong version of the aqualink.conf file.  I'd used the "Simple install; trust everything" script to install AqualinkD on a basic Raspberry Pi OS install.  I know you've predicted this to be unstable using an SD card for a boot drive, but I wanted to test to make sure I could get this working first - then I'll try to learn how to harden this system a bit with the read-only file system.

I used Geany and searched for aqualinkd.conf - found it, edited it, saved it.  But it never changed the configuration when I stoped/started Aqualinkd.  When I found the file by the same name in the etc/ folder, I edited it, saved it, restarted and the changes took effect.  I've successfully edited my button names now!   I've written this out in case someone else does the same thing in the future (unlikely, I'm a noob and at my technical ceiling.  I suspect others may not make the same mistake.)

I have two remaining questions:

One function I can't seem to find is my spa blower.  It works on the pool using the hardwired controls, but I can't seem to figure out which button in Aqualinkd should be associated that that relay.  The default button 5 was listed as Spa Blower but actually controlled my light.  Unfortunately, button 6 (formerly indicated as "Pool Light") doesn't seem to operate my blower.

It occurred to me if I use Aqualinkd to replace my wall controller, I still need to change programming on occasion.  Is there a UI to access the programming to adjust start/stop times of the pumps?

**Comment by sfeakes** [comment]
On removing the keypad completely.  There is no good solution in AqualinkD to program the timers on the control panel.  I did have a keypad simulator at one time, but it doesn’t work well with the new architecture in the last few releases of AqualinkD.  AqualinkD is designed to be integrated with a home automation hub, so I usually recommend removing all programming from the pool controller and let your home automation hub to the pump timing, it a LOT easier to manage that way. If you don’t have a hub, you can use the cron on the pi to automate the pumps etc.  (I need to document how to do that). But if you know how to use cron and read the wiki API doc is very simple to do.   At some point in the future I’ll spend time on creating a UI for this rather than getting the keypad simulator working.

on the missing spa blower, find out what key is that on the keypad, then that’s the key # to use in the config.

on the ro root.  That’s still relevant, but far less so on modern Pi and with latest OS.  It was really a hardware issue (on the Broadcom chip Pi’s used to use.  So once you get everything running, maybe just make a clone of the CF card and keep that safe.   Or if using USB to boot, it shouldn’t be a problem.

**Comment by sfeakes** [comment]
Just updating this.
Release 2.3.0 now has a scheduler and the simulator is back.  It's a pre-release at the moment

**Reply by justin-lake** [reply]
Oh this is awesome!  Just had my pool remodeled and have a new Pi Zero 2 W in transit (yes, I overpaid for it, but still better than a new Jandy control pad).  Thanks for investing time to make this happen.
